test/jacobi.c

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2018 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+#include <openssl/bn.h>
+#include "testutil.h"
+
+static int test_jacobi(unsigned int a, unsigned int b, int result)
+{
+    BIGNUM *A, *B;
+    int j;
+    int ret = 0;
+
+    A = BN_new();
+    B = BN_new();
+    if (A == NULL || B == NULL)
+        return 0;
+
+    if (!BN_set_word(A, a) || !BN_set_word(B, b)) {
+        goto err;
+    }
+
+    j = BN_jacobi_symbol(A, B, NULL);
+    if (j != result)
+        TEST_info("Jacobi test failed: (%d/%d) = %d, expected %d\n",
+                  a, b, j, result);
+    else
+        ret = 1;
+
+err:
+    BN_free(A);
+    BN_free(B);
+    return ret;
+}
+
+static int test_jacobi_symbol(void)
+{
+    int ret = 1;
+
+    ret &= test_jacobi(1, 2, -2);
```

Comment:
Wouldn't the usual way to write this be as one expression with lots of `&` operators?  Possibly eliminating the `ret` local?

I don't mind either way, so pick whichever.

Reply:
This sort of accumulator is used in other places in the tree, IIRC.  (Mostly constant-time code and tests?)

Reply:
I could also move the 3 values to a table, and write a for loop for it.